actions/channel_actions.jsx

```diff
@@ -113,34 +109,39 @@ export function searchMoreChannels(term) {
     };
 }
 
-export async function autocompleteChannels(term, success, error) {
-    const state = doGetState();
-    const teamId = getCurrentTeamId(state);
-    if (!teamId) {
-        return;
-    }
-
-    const {data, error: err} = await ChannelActions.autocompleteChannels(teamId, term)(doDispatch, doGetState);
-    if (data && success) {
-        success(data);
-    } else if (err && error) {
-        error({id: err.server_error_id, ...err});
-    }
+export function autocompleteChannels(term, success, error) {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const teamId = getCurrentTeamId(state);
+        if (!teamId) {
+            return;
+        }
+
+        const {data, error: err} = await dispatch(ChannelActions.autocompleteChannels(teamId, term));
+        if (data && success) {
+            success(data);
+        } else if (err && error) {
+            error({id: err.server_error_id, ...err});
+        }
+    };
```

Comment:
I think we should move here those that access `store` (getState and dispatch) from ``ChannelMentionProvider`` (on success).
Please also consider the one at ``ChannelProvider``.

Reply:
Hi @saturninoabril , I am not fully understanding what you mean here. I am kindly requesting that you elaborate more on `those` that I should move.

Reply:
I'm referring to the following:
- [getMyChannelMemberships(store.getState())](https://github.com/mattermost/mattermost-webapp/blob/95730d6517abf279f3d3bf38f4b62841bc1cf134/components/suggestion/channel_mention_provider.jsx#L161)
- [getChannel(store.getState(), item.id)](https://github.com/mattermost/mattermost-webapp/blob/95730d6517abf279f3d3bf38f4b62841bc1cf134/components/suggestion/channel_mention_provider.jsx#L177)
- [store.dispatch()](https://github.com/mattermost/mattermost-webapp/blob/95730d6517abf279f3d3bf38f4b62841bc1cf134/components/suggestion/channel_mention_provider.jsx#L205)

But I think that should be dealt with separately.